driver-core/src/main/java/com/datastax/driver/core/SessionManager.java

```diff
@@ -692,6 +699,12 @@ else if (fetchSize != Integer.MAX_VALUE)
    * and handle host failover.
    */
   void execute(final RequestHandler.Callback callback, final Statement statement) {
+    if (this.isClosed()) {
+      throw new IllegalStateException("Could not send request, session is closed");
```

Comment:
Just reminder that this should not throw the exception, but rather fail the future (the callback in this case I suppose)